Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -191,10 +191,12 @@ ALWAYS_INLINE bool ContainerNode::removeNodeWithScriptAssertion(Node& childToRem
         ChildListMutationScope(*this).willRemoveChild(childToRemove);
     }
 
-    ASSERT_WITH_SECURITY_IMPLICATION(ScriptDisallowedScope::InMainThread::isEventDispatchAllowedInSubtree(childToRemove));
     if (source == ChildChange::Source::API) {
         childToRemove.notifyMutationObserversNodeWillDetach();
-        dispatchChildRemovalEvents(protectedChildToRemove);
+        if (!document().shouldNotFireMutationEvents()) {
+            RELEASE_ASSERT_WITH_SECURITY_IMPLICATION(ScriptDisallowedScope::InMainThread::isEventDispatchAllowedInSubtree(childToRemove));
```

Comment:
This release assert is useless given isEventDispatchAllowedInSubtree() trivially returns true when `ASSERT_ENABLED || ENABLE(SECURITY_ASSERTIONS)` is false. We should have kept it as a debug assert instead.

Reply:
Addressing this and other issues in https://github.com/WebKit/WebKit/pull/25095

Reply:
(Meta: The thumbs-up on the comments here are just ACKs from me that I read and understood the comments…)